raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/MediaTypeInjectionPhase.java

```diff
@@ -51,6 +50,18 @@ public Node apply(Node tree)
                 {
                     injectMediaTypes(bodyNode, defaultMediaTypes);
                 }
+                else
+                {
+                    // if hasExplicitMimeType and body node has attributes type and display name, they get removed because they where incorrectly merge. See #498
+                    Node bodyNodeValue = bodyNode.getValue();
+                    Node type = NodeSelector.selectFrom("type", bodyNodeValue);
+                    Node displayName = NodeSelector.selectFrom("displayName", bodyNodeValue);
+
+                    if (type != null)
+                        bodyNodeValue.removeChild(type.getParent());
+                    if (displayName != null)
+                        bodyNodeValue.removeChild(displayName.getParent());
+                }
```

Comment:
Removing those nodes seems very risky. How hard is to modify the merging (or media type injection) instead of removing the nodes after they were applied?

Reply:
working on that